### PR TITLE
Add version view for events & locations

### PR DESCRIPTION
**Pull request by MizukiTemma**
### Short description

This PR adds version view for POI.

### Proposed changes

- Make view and template of page version view universal for page and POI

### Side effects

- <s>Note: this PR is still in progress but openred as draft to share progress etc with #2267 </s>
- If this approach looks reasonable, it will be extented for event.

### Resolved issues

Fixes: #1154
Fixes: #1155 

__________________________________________________

[Pull Request Review Guidelines](https://digitalfabrik.github.io/integreat-cms/pull-request-review-guide.html)

### CODE REVIEW

File: integreat_cms/cms/templates/pages/page_revisions.html
Comment (JoeyStk) [inline integreat_cms/cms/templates/pages/page_revisions.html:14]:
```suggestion
           class="btn btn-outline">
```
Small rebasing error

File: integreat_cms/cms/views/pages/page_revision_view.py
Comment (JoeyStk) [inline integreat_cms/cms/views/pages/page_revision_view.py:176]:
```suggestion
            return redirect_to_content_revisions
```

**Review by JoeyStk — CHANGES_REQUESTED** [review]
Thank you very much for the PR and taking over the responsibility to generalize this feature. I wasn't able to work on this PR myself, but have a few suggestions that hopefully help you in the process. 

- I think it would be best to rename page_revisions to generic_revisions or something similar. Same goes for the view. 
- I was looking into the question how to handle the context. I think one solution might be to create a new function get_context, that finds out the content type via the kwargs and then returns the navigation item that should be highlighted in the navigation bar

**Review by JoeyStk — APPROVED** [review]
To me this looks very good now. The only thing I'm not entirely sure about is, whether or not we should specify the content type in ["Back to form"](https://github.com/digitalfabrik/integreat-cms/pull/2268/files#diff-f99ce2980aa8c5acb976c0faae732414c0e70d66f329ac72fd22dd2f4ba4b7cfR14). To me either way is fine, but maybe we should get a second opinion on this

File: integreat_cms/cms/views/content_revision_view.py
Comment (timoludwig) [inline integreat_cms/cms/views/content_revision_view.py:30]: No need to define a custom method here, there's already the inbuilt [`get_context_data()`](https://docs.djangoproject.com/en/4.2/topics/class-based-views/generic-display/#adding-extra-context)

File: integreat_cms/cms/views/content_revision_view.py
Comment (timoludwig) [inline integreat_cms/cms/views/content_revision_view.py:38]: I like this idea, however there's already an inbuilt Mixin that handles permission in class-based views: [`PermissionRequiredMixin`](https://docs.djangoproject.com/en/4.2/topics/auth/default/#the-permissionrequiredmixin-mixin)

File: integreat_cms/cms/views/content_revision_view.py
Comment (timoludwig) [inline integreat_cms/cms/views/content_revision_view.py:181]: I think it would make more sense to add a `model` attribute to the sub-classes and derive the content type and name from it...

File: integreat_cms/cms/views/content_revision_view.py
Comment (timoludwig) [inline integreat_cms/cms/views/content_revision_view.py:188]: This query could be handled by the [`SingleObjectMixin`](https://docs.djangoproject.com/en/3.2/ref/class-based-views/mixins-single-object/#singleobjectmixin)

File: integreat_cms/cms/templates/content_revisions.html
Comment (timoludwig) [inline integreat_cms/cms/templates/content_revisions.html:13]: Quotes are missing and `{{ form }}` usually holds an actual form object and not a URL, so I would rename this variable.

File: integreat_cms/cms/templates/content_revisions.html
Comment (timoludwig) [inline integreat_cms/cms/templates/content_revisions.html:17]: Same here, the quotes are missing and `revision` is a bit ambiguous.

File: integreat_cms/cms/templates/content_revisions.html
Comment (timoludwig) [inline integreat_cms/cms/templates/content_revisions.html:24]: While we're at it, we could also refactor the existing template and make it more performant - if we call `{{ content_translations.count }}` several times in a template, it will make a new sql query each time. It's better to force the evaluation into a python object and call `{{ content_translations|length }}` each time, which will do the computation in Python instead of SQL and does not retrieve the objects from the database again.

File: integreat_cms/cms/templates/content_revisions.html
Comment (timoludwig) [inline integreat_cms/cms/templates/content_revisions.html:111]: The `cms.change_{{ content_name }}_object` permission only exists for pages, so we need to use a different syntax for all other content models.

File: integreat_cms/cms/views/content_revision_view.py
Comment (timoludwig) [inline integreat_cms/cms/views/content_revision_view.py:155]: There are quite a few duplications in `get()` and `post()`, whenever you notice this you can extract pieces of code into `dispatch()` which will be called in every case.

**Review by timoludwig — CHANGES_REQUESTED** [review]
Thanks a lot, really cool! :rocket: 

I think generalizing this stuff is the right way to approach things, if this goes well we should also think about generalizing other views (e.g. side by side views), it really makes things easier when these things are at a single place and we don't have to edit the same thing for all content types.

Apart from my comments, I also took a more radical approach on refactoring this stuff, e.g. making more use of Django's inbuilt view mixins etc., so I also tried to make it compatible with the imprint version view as well...
Since my changes went a bit out of hand (as always), I'm now trying a different approach and I'm going to open another PR where you can have a look at the changes before merging them into your branch :sweat_smile:

**Review by timoludwig — APPROVED** [review]
Thanks a lot! :tada:

**Pull request merged**
